libnavui-dropin/src/main/java/com/mapbox/navigation/dropin/component/routeline/RouteLineComponent.kt

```diff
@@ -4,91 +4,89 @@ import android.util.Log
 import com.mapbox.android.gestures.Utils
 import com.mapbox.geojson.Point
 import com.mapbox.maps.MapView
-import com.mapbox.maps.plugin.gestures.OnMapClickListener
-import com.mapbox.maps.plugin.gestures.gestures
-import com.mapbox.maps.plugin.locationcomponent.OnIndicatorPositionChangedListener
-import com.mapbox.maps.plugin.locationcomponent.location
+import com.mapbox.maps.Style
 import com.mapbox.navigation.base.ExperimentalPreviewMapboxNavigationAPI
 import com.mapbox.navigation.core.MapboxNavigation
-import com.mapbox.navigation.core.directions.session.RoutesObserver
-import com.mapbox.navigation.core.lifecycle.MapboxNavigationApp
 import com.mapbox.navigation.core.lifecycle.MapboxNavigationObserver
-import com.mapbox.navigation.core.trip.session.RouteProgressObserver
+import com.mapbox.navigation.dropin.component.map.MapEventDelegate
+import com.mapbox.navigation.dropin.component.routefetch.MapboxDropInRouteRequester
+import com.mapbox.navigation.dropin.extensions.flowRouteProgress
+import com.mapbox.navigation.dropin.extensions.flowRoutesUpdated
 import com.mapbox.navigation.ui.maps.route.line.MapboxRouteLineApiExtensions.findClosestRoute
 import com.mapbox.navigation.ui.maps.route.line.MapboxRouteLineApiExtensions.setRoutes
 import com.mapbox.navigation.ui.maps.route.line.api.MapboxRouteLineApi
 import com.mapbox.navigation.ui.maps.route.line.api.MapboxRouteLineView
 import com.mapbox.navigation.ui.maps.route.line.model.MapboxRouteLineOptions
 import com.mapbox.navigation.ui.maps.route.line.model.RouteLine
 import com.mapbox.navigation.utils.internal.InternalJobControlFactory
+import com.mapbox.navigation.utils.internal.ifNonNull
 import kotlinx.coroutines.cancelChildren
+import kotlinx.coroutines.flow.collect
 import kotlinx.coroutines.launch
 
 @OptIn(ExperimentalPreviewMapboxNavigationAPI::class)
 internal class RouteLineComponent(
     private val mapView: MapView,
     private val options: MapboxRouteLineOptions,
+    private val mapEventDelegate: MapEventDelegate,
```

Comment:
🤔 if it's possible that the `mapView` in the constructor vs `mapView` inside `mapEventDelegate` can be different instances? Is it ok if they are different? Why or why not? Should we rather make the `mapView` inside `mapEventDelegate` `internal`, use it from there and remove mapView from the constructor?

Reply:
The `RouteLineComponent` is dependent on the `MapView` (although I want to try and remove it) therefore it's passed via the constructor.  Providing references to the `MapView` is not the role of the `MapEventDelegate`.  All of an object's dependencies should be explicitly passed via its constructor.

Reply:
Hmm. It would be great to remove the dependency on `mapView` from `RouteLineComponent` and rather make it dependent on `MapboxMap` if required.